Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitClient.swift

```diff
@@ -150,26 +179,31 @@ import BraintreeCore
         if let error = error {
             switch error {
             case ASWebAuthenticationSessionError.canceledLogin:
-                self.apiClient.sendAnalyticsEvent("ios.sepa-direct-debit.web-flow.canceled")
+                if !returnedToAppAfterPermissionAlert {
+                    self.apiClient.sendAnalyticsEvent(BTSEPADirectAnalytics.challengeAlertCanceled)
```

Comment:
Can we add the same comment that we have for the PayPalClient? The whole `returnedToAppAfterPermissionAlert` code is confusing and it helps future readers